src/main/java/edu/illinois/cs/dt/tools/utility/iFixPlusPlugin.java

```diff
@@ -0,0 +1,744 @@
+            return null;
+        }
+    }
+
+    private String lastPolluter() {
+        if (replayPath2.toString().equals("")) {
+            return lastPolluter_full();
+        }
+        else {
+            return lastPolluter_minimized();
+        }
+    }
+
+    private String lastPolluter_full() {
+        try {
+            List<String> failorder = testFailOrder_full();
+            return failorder.get(failorder.size()-2);
+        } catch (Exception e) {
+            System.out.println("Encounter exception in lastPolluter_full!");
+            return null;
+        }
+    }
+
+    private String lastPolluter_minimized() {
+        try {
+            List<PolluterData> polluters = new Gson().fromJson(FileUtil.readFile(replayPath2), MinimizeTestsResult.class).polluters();
+            for(PolluterData pd: polluters) {
+                if (pd.deps().size() >=1) {
+                    return pd.deps().get(pd.deps().size()-1);
+                }
+            }
+            return null;
+        } catch (Exception e) {
+            System.out.println("Encounter exception in reading json for failing order!");
+            return null;
+        }
+    }
+
+    int countDirNums(String path) {
+        File [] list = new File(path).listFiles();
+        int num = 0;
+        for (File file : list) {
+            if (file.isDirectory()) {
+                num ++;
+            }
+        }
+        return num;
+    }
+
+    public void diffing() {
+        try {
+            diffSub();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void recordsubDiff(String beforeState,
+                               String afterState, String fileName) {
+        try {
+            // create a string builder
+            StringBuilder sb = new StringBuilder();
+            Diff diff = DiffBuilder.compare(beforeState).withTest(afterState).
+                    withNodeMatcher(new DefaultNodeMatcher(
+                            ElementSelectors.byName
+                    ))
+                    .checkForSimilar()
+                    .build();
+            Iterable<Difference> differences = diff.getDifferences();
+            for (Object object : differences) {
+                Difference difference = (Difference)object;
+
+                sb.append("***********************\n");
+                sb.append(difference);
+                sb.append("\n~~~~\n");
+                makeSubDifferenceReport(difference, sb);
+                sb.append("***********************\n");
+            }
+            writeToFile(fileName, sb.toString(), true);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String readFile(String path) throws IOException {
+        File file = new File(path);
+        return FileUtils.readFileToString(file, "UTF-8");
+    }
+
+    Set<String> File2SetString(String path) {
```

Comment:
Method should not start with capitalized letter, should have access modifier like `private`, maybe change to something like `readFileContentsAsSet`.